contracts/mocks/MockRewarder.sol

```diff
@@ -52,15 +52,54 @@ contract MockRewarder is IRewarder {
 
     function handleRewardToken(
         address token,
-        uint32,
+        uint32 solverId,
         uint256 amount
-    ) external override {
-        uint256 balance = IERC20(token).balanceOf(msg.sender);
-        require(balance == 0, "Escrow not empty");
-        
-        IERC20(token).safeTransfer(tx.origin, amount);
+    ) external {
+
+    }
+
+    function calculateSeekerTax(uint256 _paymentAmount)
+        public
+        view
+        returns (uint256 platformTax_, uint256 referralTax_)
+    {
+        ITaxManager taxManager = getTaxManager();
+        (platformTax_, referralTax_) = _calculateSeekerTax(taxManager, _paymentAmount);
+    }
+
+    function _calculateSeekerTax(ITaxManager _taxManager, uint256 _paymentAmount) 
+        internal 
+        view
+        returns 
+    (
+        uint256 platformTax_, 
+        uint256 referralTax_
+    ){
+        ITaxManager.SeekerFees memory seekerFees = _taxManager.getSeekerFees();
+        uint256 taxRateDivisor = _taxManager.taxBaseDivisor();
+
+        referralTax_ = (_paymentAmount * seekerFees.referralRewards) /
+            taxRateDivisor;
 
-        emit RewardClaimed(tx.origin, msg.sender, amount);
+        platformTax_ = (_paymentAmount * seekerFees.platformRevenue) /
+            taxRateDivisor;
+
+        return (referralTax_, platformTax_);
+    }
+
+    function handleSeekerTaxNative(
+        uint32 solverId,
+        uint256 referralTax,
+        uint256 platformTax
```

Comment:
needs to be implemented soon (not this PR)

Reply:
This is for the mockRewarder contract, it has already been implemented in the actual Rewarder.sol